### ISSUE TITLE
Gradient Normalization when using max-sentences

### ISSUE THREAD

**Issue by ajesujoba** [pytorch/fairseq#2192]
Does fairseq normalize gradient with the number of tokens in a batch when I specify just the `max-sentence ` without `--sentence-avg` and `max-token`? 

If no, is there a way I can do that?

**Comment by myleott** [comment]
Yes, by default it will normalize by the number of tokens, even when you specify the batch size with `--max-sentence`.
